marshmallow_dataclass/__init__.py

```diff
@@ -383,18 +403,34 @@ def field_for_schema(
     origin = typing_inspect.get_origin(typ)
     if origin:
         arguments = typing_inspect.get_args(typ, True)
+        generic_type_mapping = {
+            **DEFAULT_GENERIC_TYPE_MAPPING,
+            **(base_schema.TYPE_MAPPING if base_schema else {}),
+        }
         if origin in (list, List):
-            child_type = field_for_schema(arguments[0], base_schema=base_schema)
-            return marshmallow.fields.List(child_type, **metadata)
+            child_type = field_for_schema(
+                arguments[0], base_schema=base_schema
+            )
+            list_type = generic_type_mapping[List]
+            return list_type(child_type, **metadata)
```

Comment:
Do we really need `DEFAULT_GENERIC_TYPE_MAPPING` ? Could we just use `type_mapping.get(List, marshmallow.fields.List)` ?

Reply:
I don't mind using `type_mapping.get(List, marshmallow.fields.List)`, I just thought it was simpler to understand having a single location with the defaults rather than having to dive in the code to figure out what defaults are used.

But it's up to you, say the word and I remove `DEFAULT_GENERIC_TYPE_MAPPING` =]

Reply:
I understand it's a tradeoff, but I think the code will be both easier to read and more performant with `.get(key, default)`

Reply:
ok, updated 👍